common/ayon_common/distribution/file_handler.py

```diff
@@ -33,14 +34,15 @@ def download_url(
                 hops allowed
             headers (Optional[dict[str, str]]): Additional required headers
                 - Authentication etc..
-        """
 
+        """
         root = os.path.expanduser(root)
         if not filename:
             filename = os.path.basename(url)
         fpath = os.path.join(root, filename)
 
-        os.makedirs(root, exist_ok=True)
+        if not os.path.exists(root):
```

Comment:
Is this necessary when there is flag `exist_ok` ?

Reply:
Yes for case the folder exists but user does not have permissions.